opnode/test/system_test.go

```diff
@@ -413,3 +415,25 @@ func TestMissingBatchE2E(t *testing.T) {
 	require.Nil(t, err, "Get block from sequencer")
 	require.NotEqual(t, block.Hash(), receipt.BlockHash, "L2 Sequencer did not reorg out transaction on it's safe chain")
 }
+
+// TestSystemMockP2P sets up a L1 Geth node, a rollup node, and a L2 geth node and then confirms that
+// the nodes can sync L2 blocks before they are confirmed on L1.
+func TestSystemMockP2P(t *testing.T) {
+	log.Root().SetHandler(log.DiscardHandler()) // Comment this out to see geth l1/l2 logs
+
+	cfg := defaultSystemConfig(t)
+	// slow down L1 blocks so we can see the L2 blocks arrive before the L1 blocks do.
+	cfg.L1BlockTime = 10
+	// connect the nodes
+	cfg.P2PTopology = map[string][]string{
+		"verifier": []string{"sequencer"},
+	}
+
+	sys, err := cfg.start()
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	t.Log("successfully set up network")
+
+	// TODO: await L2 blocks going from sequencer to verifier, ahead of L1
```

Comment:
This is a good basic test. We should also test an unsafe reorg once the logic is ready.